Question(1,2,3,4)

aditya-wagh opened the pull request.

aditya-wagh committed:
Question(1,2,3,4)

pandey-adarsh147 commented on Assignment/src/question1/Item.java:
```diff
@@ -0,0 +1,102 @@
+
+  
+    
+    public double calculateTaxes(){
```
Create sub class to implement this method for various item type. This help to implement plug and play feature if new item introduced, you are not required to change anything into original class. create abstract class

pandey-adarsh147 commented on Assignment/src/question1/Item.java:
```diff
@@ -0,0 +1,102 @@
+        double taxes=0;
+        if(this.type.equals("raw"))
+        {
+            taxes=0.125* this.price;
```
Constants should be maintain in separate file.
Coding guideline not followed.
Coding formatting very bad.

pandey-adarsh147 commented on Assignment/src/question1/Item.java:
```diff
@@ -0,0 +1,102 @@
+        {
+            taxes=0.125* this.price;
+        }
+        else if(this.type.equals("manufactured")){
```
whenever compare string equality, always use constant first then variable to avoid nullpointer
e.g.

"manufactured".equals(this.type)

instead of this.type.equals("manufactured")

pandey-adarsh147 commented on Assignment/src/question1/Item.java:
```diff
@@ -0,0 +1,102 @@
+        }
+            else
+            {
+               taxes+=10;            }
```
Readability of code is important. Although, this is language feature but we should avoid this.

taxes = taxes + 10;
instead of taxes += 10;

pandey-adarsh147 commented on Assignment/src/question3/Question3.java:
```diff
@@ -0,0 +1,387 @@
+                    for (Node i : anc) {
+                        System.out.println(i.getName());
+                    }
+                    ;
```
?

pandey-adarsh147 commented on Assignment/src/question3/Question3.java:
```diff
@@ -0,0 +1,387 @@
+    }
+
+    public int addNode(String name) {
+        Node new_node = new Node(id);
```
variable name not proper

pandey-adarsh147 commented on Assignment/src/question2/Question2.java:
```diff
@@ -0,0 +1,410 @@
+                        System.out.println("Inappropriate RollNo");
+                    }
+                } catch (NumberFormatException ex) {
+                    System.out.println(error);
```
Proper message should be printed

pandey-adarsh147 replied:
End user don't know, software error

pandey-adarsh147 reviewed (changes requested):
In general
. Code formatting
. Oops concept is missing
. Constant handling is not proper
. Variable names

aditya-wagh committed:
Suggested changes are made

Following changes are made to code:
1 Use of abstract class in Question 1.
2. Used taxes = taxes + 10; instead of taxes+=10;
3. Used proper variable names
4. Proper messages printed for errors.
5. Used constants first when comparing  two strings
